Feat: Update the screen-reader-text CSS class and its local implementations

hbhalodia opened the pull request:
<!--
Hi there! Thanks for contributing to WordPress!

Pull Requests in this GitHub repository **must** be linked to a ticket in the WordPress Core Trac instance (https://core.trac.wordpress.org), and are only used for code review. **No pull requests will be merged on GitHub.**

See the WordPress Handbook page on using PRs for Code Review more information: https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/

If this is your first time contributing, you may also find reviewing these guides first to be helpful:
- FAQs for New Contributors: https://make.wordpress.org/core/handbook/tutorials/faq-for-new-contributors/
- Contributing with Code Guide: https://make.wordpress.org/core/handbook/contribute/
- WordPress Coding Standards: https://make.wordpress.org/core/handbook/best-practices/coding-standards/
- Inline Documentation Standards: https://make.wordpress.org/core/handbook/best-practices/inline-documentation-standards/
- Browser Support Policies: https://make.wordpress.org/core/handbook/best-practices/browser-support/
- Proper spelling and grammar related best practices: https://make.wordpress.org/core/handbook/best-practices/spelling/
-->

<!-- Insert a description of your changes here -->

Trac ticket: https://core.trac.wordpress.org/ticket/62238

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

hbhalodia committed:
Update the css for screen reader text

afercia reviewed (dismissed):
Thanks @hbhalodia 
A couple first point for feedback:
- Some stylesheet meant to support old browsers like Internet Explorer 7 must keep the old `clip`.
- In other cases where `clip` is removed and there's no `clip-path`, then `clip-path` should be added.

hbhalodia commented:
> Thanks @hbhalodia A couple first point for feedback:
> 
> * Some stylesheet meant to support old browsers like Internet Explorer 7 must keep the old `clip`.
> * In other cases where `clip` is removed and there's no `clip-path`, then `clip-path` should be added.

Hi @afercia, thanks for the feedback. I would resolve this.

hbhalodia committed:
Revert clip for ie and add clip path instead of clip

hbhalodia commented:
Hello @afercia, I have updated the PR with the requested changes.

Thank You,

audrasjb commented on src/wp-content/themes/twentyfourteen/css/ie.css:
```diff
@@ -1332,4 +1332,4 @@ a.post-thumbnail:hover {
 	margin-left: 190px;
-}
\ No newline at end of file
+}
```
this change doesn't look relevant to the scope of the ticket :)

hbhalodia replied:
This is now resolved.

audrasjb commented on src/wp-content/themes/twentytwelve/css/ie.css:
```diff
@@ -15,10 +15,12 @@ body.custom-background-white .site {
 	margin-top: 0;
 	padding: 0;
 }
+
```
these changes don't look relevant given the scope of the ticket :)

hbhalodia replied:
This is now resolved.

hbhalodia committed:
Merge branch 'trunk' into fix/issue-62238

hbhalodia committed:
Remove the unscoped change in PR

audrasjb commented on src/wp-content/themes/twentytwelve/css/ie.css:
```diff
@@ -270,4 +271,4 @@ input[type="password"] {
 	z-index: 1;
-}
\ No newline at end of file
+}
```
These changes look unrelated :)

hbhalodia replied:
Thanks @audrasjb, I would look all this unrelated changes and remove/revert them in next commit.

hbhalodia replied:
This is now resolved.

Thank You,

audrasjb reviewed (dismissed):
There are still some unrelated changes

hbhalodia committed:
Remove unrelated changes from the PR scope related to linting

joedolson committed:
Add coverage in a few additional cases

These are screen reader hidden texts using other classes. For the older themes, I left support for `clip` with a conment.

joedolson reviewed (approved):
I added coverage for a handful of additional cases, and with that I think this is pretty ready. There may be other things missed, but we shouldn't hold this up because of that.

joedolson reviewed (changes requested):
This is not actually ready; instances of clip: auto that are used to render screen reader text visible also need to be updated.

joedolson committed:
Handle :focus states of screen reader text

joedolson commented:
Updated `:focus` states to not use `clip` if the related static state no longer uses `clip`.

hbhalodia commented:
Thanks @joedolson, to update the PR with required changes 🙇.

joedolson committed:
Additional change in theme.php

joedolson committed:
Additional changes in themes.

joedolson committed:
Rebuild Twenty Twenty One CSS

joedolson committed:
Also update in theme customizer JS.

joedolson committed:
Additional change in twentyeleven

joedolson committed:
No longer needed

joedolson committed:
Remove clip reset in style; restore in ie.css

joedolson commented:
In r59832
